crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1748,8 +1748,37 @@ impl<'db> TypeInferenceBuilder<'db> {
         self.infer_arguments(arguments);
         self.infer_expression(func);
 
-        // TODO resolve to return type of `func`, if its a callable type
-        Type::Unknown
+        match func.as_ref() {
+            name_expr @ ast::Expr::Name(..) => {
+                let name_expr_id = name_expr.scoped_ast_id(self.db, self.scope);
+                let Some(named_type) = self.types.expressions.get(&name_expr_id) else {
+                    return Type::Unbound;
+                };
+
+                match named_type {
```

Comment:
I think we will want to follow the existing pattern of `Type::member` here, with something like `Type::call`, so this match on `Type` variants happens inside `Type` rather than in type inference (the code here would reduce to basically a one liner like `func_ty.call()`). The main reason for this is that the definition of `call` for unions and intersections (which doesn't have to be implemented in this PR) will involve recursively calling `call` on the elements of the union/intersection.

Reply:
Ahhhh, that makes sense. 👍🏼 Will move this over there!